### PR TITLE
Added tests for KubernetesClient serialization/deserialization

### PR DESCRIPTION
**Pull request by manusa**
Should ensure that (de)serialization works as expected both in JVM and Native modes.

Relates to #39934

/cc @galderz

### CODE REVIEW

**Review by gsmet — APPROVED** [review]
Thanks!

File: integration-tests/kubernetes-client/src/test/java/io/quarkus/it/kubernetes/client/KubernetesClientSerializationTest.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+@WithKubernetesTestServer(setup = KubernetesClientSerializationTest.CrudEnvironmentPreparation.class)
+@QuarkusTest
+public class KubernetesClientSerializationTest {
+
+    @KubernetesTestServer
+    private KubernetesServer mockServer;
+
+    @Test
+    @DisplayName("PUT with String body - Should serialize and persist the Pod")
+    void serialization() {
```
Comment (galderz) [inline integration-tests/kubernetes-client/src/test/java/io/quarkus/it/kubernetes/client/KubernetesClientSerializationTest.java:40]: I'm not sure this is really testing what I was looking for. In the problematic test we start by building a `Pod` instance and how that gets transform into the actual JSON is the problematic area. Is this really being tested here?

Comment (galderz) [reply integration-tests/kubernetes-client/src/test/java/io/quarkus/it/kubernetes/client/KubernetesClientSerializationTest.java:40]: Ah never mind, it's deserialization it seems

File: integration-tests/kubernetes-client/src/test/java/io/quarkus/it/kubernetes/client/KubernetesClientSerializationTest.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                                .withName("deserialization-test-container")
+                                .addToPorts(new ContainerPortBuilder()
+                                        .withContainerPort(8080)
+                                        .build())
+                                .build())
+                        .build())
+                .build())
+                .create();
+        // JSON data is retrieved from the mock server
+        when()
```
Comment (galderz) [inline integration-tests/kubernetes-client/src/test/java/io/quarkus/it/kubernetes/client/KubernetesClientSerializationTest.java:102]: Is this really verifying that the JSON shape is the expected one and not the one with all the empty objects/arrays? https://github.com/quarkusio/quarkus/issues/39934#issuecomment-2119773533

Comment (manusa) [reply integration-tests/kubernetes-client/src/test/java/io/quarkus/it/kubernetes/client/KubernetesClientSerializationTest.java:102]: AFAIK, it should.
In any case, if you want to verify manually you can change line 113 from
```java
                .log().ifValidationFails(LogDetail.BODY);
```
to
```java
                .log().body();
```

We can also add another assertion to verify that the response body doesn't actually contain the word `overhead`:
```java
.body(not(containsString("overhead"))
```

**Pull request merged**